torchgeo/trainers/chesapeake.py

```diff
@@ -109,12 +130,42 @@ def validation_step(  # type: ignore[override]
 
         loss = self.loss(y_hat, y)
 
+        # by default, the test and validation steps only log per *epoch*
         self.log("val_loss", loss)
-        self.log("val_acc_step", self.val_accuracy(y_hat_hard, y))
+        self.val_accuracy(y_hat_hard, y)
+        self.val_iou(y_hat_hard, y)
+
+        if batch_idx < 10:
+            # Render the image, ground truth mask, and predicted mask for the first
+            # image in the batch
+            img = np.rollaxis(  # convert image to channels last format
+                batch["image"][0].cpu().numpy(), 0, 3
+            )
+            mask = batch["mask"][0].cpu().numpy()
+            pred = y_hat_hard[0].cpu().numpy()
+            fig, axs = plt.subplots(1, 3, figsize=(12, 4))
+            axs[0].imshow(img[:, :, :3])
+            axs[0].axis("off")
+            axs[1].imshow(mask, vmin=0, vmax=6, cmap=CMAP, interpolation="none")
+            axs[1].axis("off")
+            axs[2].imshow(pred, vmin=0, vmax=6, cmap=CMAP, interpolation="none")
+            axs[2].axis("off")
```

Comment:
Can we instead move the plotting logic to the `ChesapeakeCVPR` dataset class? That's how `RasterDataset` and `VectorDataset` are designed. Then you can just call `dataset.plot(sample)` and the code can be reused by anyone using the dataset in their own code.

Reply:
`RasterDataset` plot takes a single Tensor as input, while this code is plotting (image, mask, predictions). Do you imagine multiple versions of plot?

Reply:
Ah, I think this is the first GeoDataset we have with both `image` and `mask`. We should change `plot` to take in a sample dict instead.